hw2/task01/task01.py

```diff
@@ -9,9 +9,14 @@
     5) Find most common non ascii char for document
 """
 
-import string
+import sys
+from typing import Generator
 from unicodedata import category
 
+codepoints = range(sys.maxunicode + 1)
+punctuation = {c for i in codepoints if category(c := chr(i)).startswith("P")}
+ASCII = {chr(x) for x in range(256)}
```

Comment:
It is better.  But it can be improved even more.  Python already has a built-in method for ASCII check, which you can use.